src/read.c

```diff
@@ -378,92 +380,92 @@ static void AssignRef(const LHSRef ref)
     }
 }
 
-static void UnbindRef(ScannerState * s, const LHSRef ref)
+static void UnbindRef(ReaderState * rs, const LHSRef ref)
 {
     TRY_IF_NO_ERROR
     {
         switch (ref.type) {
         case R_LVAR:
-            IntrUnbLVar(ref.var);
+            IntrUnbLVar(&rs->intr, ref.var);
             break;
         case R_HVAR:
-            IntrUnbHVar(ref.var);
+            IntrUnbHVar(&rs->intr, ref.var);
             break;
         case R_DVAR:
-            IntrUnbDVar(ref.var, ref.nest0);
+            IntrUnbDVar(&rs->intr, ref.var, ref.nest0);
             break;
         case R_GVAR:
-            IntrUnbGVar(ref.var);
+            IntrUnbGVar(&rs->intr, ref.var);
             break;
         case R_ELM_LIST:
-            IntrUnbList(ref.narg);
+            IntrUnbList(&rs->intr, ref.narg);
             break;
         case R_ELM_POSOBJ:
-            IntrUnbPosObj();
+            IntrUnbPosObj(&rs->intr);
             break;
         case R_ELM_REC_NAME:
-            IntrUnbRecName(ref.rnam);
+            IntrUnbRecName(&rs->intr, ref.rnam);
             break;
         case R_ELM_REC_EXPR:
-            IntrUnbRecExpr();
+            IntrUnbRecExpr(&rs->intr);
             break;
         case R_ELM_COMOBJ_NAME:
-            IntrUnbComObjName(ref.rnam);
+            IntrUnbComObjName(&rs->intr, ref.rnam);
             break;
         case R_ELM_COMOBJ_EXPR:
-            IntrUnbComObjExpr();
+            IntrUnbComObjExpr(&rs->intr);
             break;
         case R_INVALID:
         case R_ELMS_LIST:
         case R_FUNCCALL:
         case R_FUNCCALL_OPTS:
         default:
-            SyntaxError(s, "Illegal operand for 'Unbind'");
+            SyntaxError(&rs->s, "Illegal operand for 'Unbind'");
         }
     }
 }
 
-static void IsBoundRef(ScannerState * s, const LHSRef ref)
+static void IsBoundRef(ReaderState * rs, const LHSRef ref)
```

Comment:
Same as `EvalRef`; note that the `ScannerState` is contained in the `ReaderState`, so there is no need for us to pass both.